ui/src/app/shared/header/chips/filter-chip.component.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Component, ElementRef, EventEmitter, Input, OnInit, Output,
+  ViewChild
+} from "@angular/core";
+import {FieldDataType} from "../../common";
+import {URLSearchParamsUtils} from "../../utils/url-search-params.utils";
+import {CapabilitiesService} from "../../../core/capabilities.service";
+import {MatMenuTrigger} from "@angular/material";
+
+@Component({
+  selector: 'jm-filter-chip',
+  templateUrl: './filter-chip.component.html',
+  styleUrls: ['./filter-chip.component.css'],
+})
+export class FilterChipComponent implements OnInit {
+  @Input() chipKey: string;
+  @Input() initialChipValue: string;
+  @Output() updateValue: EventEmitter<string> = new EventEmitter();
+  @Output() removeChip: EventEmitter<null> = new EventEmitter();
+
+  @ViewChild('freeTextInput') freeTextInput: ElementRef;
+  @ViewChild(MatMenuTrigger) chipMenuTrigger: MatMenuTrigger;
+
+  currentChipValue: string;
+  options: Map<string, FieldDataType>;
+
+  constructor(
+    private readonly capabilitiesService: CapabilitiesService,
+  ) { }
+
+  ngOnInit(): void {
+    this.options = URLSearchParamsUtils.getQueryFields(
+      this.capabilitiesService.getCapabilitiesSynchronous());
+    this.currentChipValue = this.initialChipValue;
+  }
+
+  focusInput() {
+    if (this.freeTextInput) {
+      setTimeout(() => this.freeTextInput.nativeElement.focus(), 100);
```

Comment:
The `100` isn't ideal, does 0 not work? If it's needed, could you add a comment here explaining?

Reply:
Oops, this was a leftover workaround from an earlier bug with multiple triggers. Removed.